vms/proposervm/vm.go

```diff
@@ -236,6 +239,45 @@ func (vm *VM) Shutdown(ctx context.Context) error {
 	return vm.ChainVM.Shutdown(ctx)
 }
 
+func (vm *VM) NewHTTPHandler(ctx context.Context) (http.Handler, error) {
+	// Get inner VM's HTTP handler
+	innerHandler, err := vm.ChainVM.NewHTTPHandler(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the multiplexer
+	mux := http.NewServeMux()
+
+	// Add ProposerVM specific handlers
+	service := &service{vm: vm}
+	proposerVMPath, proposerVMHandler := proposervmconnect.NewProposerVMHandler(service)
+	vm.ctx.Log.Info("Registering ProposerVM Connect handler", zap.String("path", proposerVMPath))
+	mux.Handle(proposerVMPath, proposerVMHandler)
+
+	// Add gRPC reflection
+	reflectionPattern, reflectionHandler := grpcreflect.NewHandlerV1(
+		grpcreflect.NewStaticReflector(proposervmconnect.ProposerVMName),
+	)
+	vm.ctx.Log.Info("Registering gRPC reflection handler", zap.String("pattern", reflectionPattern))
+	mux.Handle(reflectionPattern, reflectionHandler)
+
+	// Add inner VM handler as fallback if it exists
+	if innerHandler != nil {
+		vm.ctx.Log.Info("Adding inner VM handler as fallback")
+		mux.Handle("/", innerHandler)
+	}
```

Comment:
This shadows reflection handling done by the inner VM's grpc server.

Prior to this change xsvm is able to expose its reflection logic:
```sh
~: grpcurl -plaintext -H "Avalanche-Api-Route:$CHAIN_ID" localhost:9650 xsvm.Ping.Ping
{}
```

With this change:
```sh
~: grpcurl -plaintext -H "Avalanche-Api-Route:$CHAIN_ID" localhost:9650 xsvm.Ping.Ping
Error invoking method "xsvm.Ping.Ping": target server does not expose service "xsvm.Ping"
```

I think we (josh and myself) had talked about adding another header to the `Avalanche-Api-Route` for the proposervm so that we could fully separate the two implementations.